client/v2/client.go

```diff
@@ -86,12 +98,28 @@ func NewClient(conf Config) Client {
 }
 
 // NewUDPClient returns a client interface for writing to an InfluxDB UDP
-// service address, specified by addr. addr should be of the form
-// "host:port" or "[ipv6-host%zone]:port". Write will fail if addr is invalid.
-func NewUDPClient(addr string) Client {
-	return &udpclient{
-		addr: addr,
+// service from the given config.
+func NewUDPClient(conf UDPConfig) (Client, error) {
+	var udpAddr *net.UDPAddr
+	udpAddr, err := net.ResolveUDPAddr("udp", conf.Addr)
```

Comment:
Yes, moving this to the New() (or perhaps an Open()) call is absolutely the right thing to do. We should have caught this earlier.